crates/api/src/transport.rs

```diff
@@ -139,6 +141,7 @@ pub trait TxImp: 'static + Send + Sync + std::fmt::Debug {
 pub type DynTxImp = Arc<dyn TxImp>;
 
 /// A high-level wrapper around a low-level [DynTxImp] transport implementation.
+#[automock]
```

Comment:
```suggestion
#[cfg_attr(any(test, feature = "test_utils"), automock)]
```

?

Reply:
Yes, applied. `all(test, feature = "test_utils")` doesn't work. Do you know why? I would think that automock only works in a test when `test_utils` is enabled.

Reply:
Do we want to use `test-utils` rather than `test_utils` this time? Or no preference?

Reply:
I wish rust just never allowed dashes anywhere in filenames or features or anything... then everything could be underscores and we wouldn't have to worry about stuff like this : ) -- that being said, given the way things are already, I don't have a preference. Totally cool to switch this or not : )

Reply:
I have no preference whatsoever, just for consistency. @ThetaSinner Are you suggesting using dashes in features?

Reply:
And would be great to know the answer to this https://github.com/holochain/kitsune2/pull/63#discussion_r1906250155.

Reply:
> Yes, applied. all(test, feature = "test_utils") doesn't work. Do you know why? I would think that automock only works in a test when test_utils is enabled.

I'm not sure about this. it's not documented so I'd expect this to behave like any other attribute and be possible to conditionally enable. What happens?

Reply:
Oh but shouldn't it be an "any"? So that it's enabled either on "test" or from another crate if "test_utils" is enabled?

Reply:
When I use `all` instead of `any` and then run `make test`, the mock imports cannot be found in the tests.

Reply:
Using the automock'd struct inside of `kitsune2_api` fails if run without feature `test_utils`.

Reply:
Here it gets wicked: With `all`, the test in `kitsune2_api` succeeds when run with `--features test-utils`. But the test in `kitsune2_core` only works with `any`.

Reply:
Mystery lifted: `#[cfg(test)]` is only active for unit tests of the crate where the tests are run. So `all(test, feature = "mockall")` will work for `kitsune2_api` if there were unit tests _and_ the feature is enabled when running the test. It will never work for tests outside of the crate, because `cfg(test)` is then inactive. That's why `any` must be used.